Respuestas a las preguntas de la Actividad 1:

¿Quién produjo originalmente los datos?

Producción original: United Nations (Organización de las Naciones Unidas - ONU), World Population Prospects (2024).  

¿Qué organización los procesa para su utilización en Our World in Data?

Fueron procesados por la organización Our World in Data (OWID).  

¿Qué representa la variable que analizarás?

Representa el cambio anual en la población (aumento o disminución absoluta de habitantes en un año determinado por entidad/país).  

In [2]:
import pandas as pd
import numpy as np

In [3]:
dataframe = pd.read_csv("../data/raw/annual-population-growth/annual-population-growth.csv")

In [4]:
dataframe.head(3)

,Entity,Code,Year,Annual change in population,Annual population change (Projected)
0,Afghanistan,AFG,1951,103163.0,NaN
1,Afghanistan,AFG,1952,108441.0,NaN
2,Afghanistan,AFG,1953,108919.0,NaN


In [5]:
dataframe.shape

(38400, 5)

In [6]:
dataframe.columns.tolist()

['Entity',
 'Code',
 'Year',
 'Annual change in population',
 'Annual population change (Projected)']

In [7]:
dataframe.dtypes

Entity                                      str
Code                                        str
Year                                      int64
Annual change in population             float64
Annual population change (Projected)    float64
dtype: object

In [8]:
dataframe.describe()

,Year,Annual change in population,Annual population change (Projected)
count,38400.000000,1.868800e+04,1.971200e+04
mean,2025.500000,2.063019e+06,8.375481e+05
std,43.300872,9.440859e+06,5.816745e+06
min,1951.000000,-3.315935e+06,-2.291035e+07
25%,1988.000000,1.781000e+03,-9.742750e+03
50%,2025.500000,4.689800e+04,9.350000e+01
75%,2063.000000,3.021128e+05,1.411160e+05
max,2100.000000,9.337137e+07,7.206736e+07


In [9]:
dataframe.isnull().sum()

Entity                                      0
Code                                     1200
Year                                        0
Annual change in population             19712
Annual population change (Projected)    18688
dtype: int64

In [10]:
dataframe["Entity"].nunique() 

256

<br>
Respuestas a las preguntas de la Actividad 2:

¿Cuántas filas y columnas tiene el dataset?
Tiene 880 filas y 5 columnas (correspondientes al fragmento/extensión procesado en el dataset).

¿Qué tipo de datos contiene la variable Year?
Contiene tipo de dato entero (int64).

¿Existen valores faltantes?
Sí. La columna Annual change in population tiene valores faltantes en los años proyectados futuros, y la columna Annual population change (Projected) tiene faltantes en el periodo histórico. La columna Code también puede tener valores nulos en entidades regionales agregadas.

¿Cuántas entidades diferentes contiene el dataset?
Contiene 10 entidades diferentes (ej. Afghanistan, Africa (UN), Albania, Algeria, American Samoa, Americas (UN), Andorra, Angola, Anguilla, Antigua and Barbuda). 

</br>

Actividad 3

In [11]:
# Filtrar los datos para el año histórico 1951
df_1951 = dataframe[dataframe["Year"] == 1951].dropna(subset=["Annual change in population"])

# Variable sobre la que aplicaremos el análisis[cite: 2]
poblacion_1951 = df_1951["Annual change in population"]

# Cálculos estadísticos[cite: 2]
media = poblacion_1951.mean()
mediana = poblacion_1951.median()
varianza = poblacion_1951.var()
desviacion_std = poblacion_1951.std()
minimo = poblacion_1951.min()
maximo = poblacion_1951.max()

# Identificar entidades del mínimo y máximo[cite: 2]
entidad_min = df_1951.loc[poblacion_1951.idxmin()]["Entity"]
entidad_max = df_1951.loc[poblacion_1951.idxmax()]["Entity"]

# Imprimir resultados
print(f"Año seleccionado: 1951")
print(f"Media: {media:,.2f}")
print(f"Mediana: {mediana:,.2f}")
print(f"Varianza: {varianza:,.2f}")
print(f"Desviación Estándar: {desviacion_std:,.2f}")
print(f"Mínimo: {minimo:,.0f} ({entidad_min})")
print(f"Máximo: {maximo:,.0f} ({entidad_max})")

Año seleccionado: 1951
Media: 1,123,856.71
Mediana: 36,355.50
Varianza: 23,946,448,116,977.24
Desviación Estándar: 4,893,510.82
Mínimo: -533,158 (North Korea)
Máximo: 43,834,182 (World)


¿Cuál es el valor de la media?

$1,123,856.71$

¿Cuál es el valor de la mediana?

$36,355.50$

¿Cuál es el valor de la desviación estándar?

$4,893,510.82$

¿Qué entidad presenta el valor mínimo y cuál el máximo?

Mínimo: North Korea ($-533,158$)

Máximo: World ($43,834,182$)

Actividad 4

In [110]:
import sys

print("Python:", sys.version)
print("Pandas:", pd.__version__)
print("NumPy:", np.__version__)

Python: 3.14.7 (v3.14.7:823f0323ee6, Aug  5 2026, 07:07:01) [Clang 21.0.0 (clang-2100.1.1.101)]
Pandas: 3.0.5
NumPy: 2.5.2


In [13]:
df_year = dataframe[["Year","Entity"]]
df_year.head()

,Year,Entity
0,1951,Afghanistan
1,1952,Afghanistan
2,1953,Afghanistan
3,1954,Afghanistan
4,1955,Afghanistan


In [14]:
df_extraccion = dataframe[dataframe["Year"]==2020]
df_extraccion.head()

,Entity,Code,Year,Annual change in population,Annual population change (Projected)
69,Afghanistan,AFG,2020,1212852.0,NaN
219,Africa (UN),UN_AFR,2020,32815675.0,NaN
369,Albania,ALB,2020,-13061.0,NaN
519,Algeria,DZA,2020,747543.0,NaN
669,American Samoa,ASM,2020,-451.0,NaN


In [15]:
df_2020_mex = dataframe[
    (dataframe["Entity"] == "Mexico") &
    (dataframe["Year"] >= 2005) &
    (dataframe["Year"] <= 2020)
][["Entity","Year","Annual change in population"]]
df_2020_mex.head()

,Entity,Year,Annual change in population
22254,Mexico,2005,1417373.0
22255,Mexico,2006,1442162.0
22256,Mexico,2007,1520690.0
22257,Mexico,2008,1599930.0
22258,Mexico,2009,1625434.0


In [16]:
df_2020_mex.mean(numeric_only=True)

Year                              2012.500
Annual change in population    1400307.625
dtype: float64

In [24]:
dataframe_new = dataframe[["Entity"]]
dataframe_new.head()

,Entity
0,Afghanistan
1,Afghanistan
2,Afghanistan
3,Afghanistan
4,Afghanistan


In [32]:
dataframe.loc[:,["Entity","Year"]]

,Entity,Year
0,Afghanistan,1951
1,Afghanistan,1952
2,Afghanistan,1953
3,Afghanistan,1954
4,Afghanistan,1955
...,...,...
38395,Zimbabwe,2096
38396,Zimbabwe,2097
38397,Zimbabwe,2098
38398,Zimbabwe,2099


In [56]:
dataframe.iloc[30:35]

,Entity,Code,Year,Annual change in population,Annual population change (Projected)
30,Afghanistan,AFG,1981,-1231727.0,NaN
31,Afghanistan,AFG,1982,-946206.0,NaN
32,Afghanistan,AFG,1983,-73395.0,NaN
33,Afghanistan,AFG,1984,272237.0,NaN
34,Afghanistan,AFG,1985,236633.0,NaN


In [60]:
dataframe.loc[10000]

Entity                                  Equatorial Guinea
Code                                                  GNQ
Year                                                 2051
Annual change in population                           NaN
Annual population change (Projected)              46407.0
Name: 10000, dtype: object

In [63]:
dataframe.loc[:,["Entity","Year","Annual change in population"]]

,Entity,Year,Annual change in population
0,Afghanistan,1951,103163.0
1,Afghanistan,1952,108441.0
2,Afghanistan,1953,108919.0
3,Afghanistan,1954,111251.0
4,Afghanistan,1955,119027.0
...,...,...,...
38395,Zimbabwe,2096,NaN
38396,Zimbabwe,2097,NaN
38397,Zimbabwe,2098,NaN
38398,Zimbabwe,2099,NaN


In [66]:
dataframe.loc[10000,["Entity","Year","Annual change in population"]]

Entity                         Equatorial Guinea
Year                                        2051
Annual change in population                  NaN
Name: 10000, dtype: object

In [76]:
dataframe.loc[
    (dataframe["Entity"] == "Mexico") &
    (dataframe["Year"] == 2020)]

,Entity,Code,Year,Annual change in population,Annual population change (Projected)
22269,Mexico,MEX,2020,1036075.0,NaN


In [80]:
dataframe.iloc[0:5]

,Entity,Code,Year,Annual change in population,Annual population change (Projected)
0,Afghanistan,AFG,1951,103163.0,NaN
1,Afghanistan,AFG,1952,108441.0,NaN
2,Afghanistan,AFG,1953,108919.0,NaN
3,Afghanistan,AFG,1954,111251.0,NaN
4,Afghanistan,AFG,1955,119027.0,NaN


In [83]:
dataframe.iloc[:,[0,2,3]]

,Entity,Year,Annual change in population
0,Afghanistan,1951,103163.0
1,Afghanistan,1952,108441.0
2,Afghanistan,1953,108919.0
3,Afghanistan,1954,111251.0
4,Afghanistan,1955,119027.0
...,...,...,...
38395,Zimbabwe,2096,NaN
38396,Zimbabwe,2097,NaN
38397,Zimbabwe,2098,NaN
38398,Zimbabwe,2099,NaN


In [85]:
dataframe.iloc[0:4,0:3]

,Entity,Code,Year
0,Afghanistan,AFG,1951
1,Afghanistan,AFG,1952
2,Afghanistan,AFG,1953
3,Afghanistan,AFG,1954


In [104]:
dataframe_filtro = dataframe.loc[
    (dataframe["Entity"] == "Honduras") &
    (dataframe["Year"] >= 2005) & 
    (dataframe["Year"] <= 2023)]

dataframe_new.head(20)

serie_poblacion = dataframe_filtro.loc[:,["Annual change in population"]]

media = serie_poblacion.mean()
mediana = serie_poblacion.median()
varianza = serie_poblacion.var()
desviacion_std = serie_poblacion.std()
minimo = serie_poblacion.min()
maximo = serie_poblacion.max()

print("Media:", media)
print("Mediana:", mediana)
print("Varianza:", varianza)
print("Desviación Estándar:",desviacion_std)
print("Mínimo:", minimo)
print("Máximo:", maximo)



Media: Annual change in population    176097.368421
dtype: float64
Mediana: Annual change in population    175977.0
dtype: float64
Varianza: Annual change in population    6.164734e+06
dtype: float64
Desviación Estándar: Annual change in population    2482.888175
dtype: float64
Mínimo: Annual change in population    170243.0
dtype: float64
Máximo: Annual change in population    180973.0
dtype: float64
